include/conf.hpp

```diff
@@ -29,11 +31,16 @@ class Config {
     uint8_t icmp;
     uint32_t payload;
 
+    /* Link-layer types */
+    uint8_t wireless;
+    uint8_t wired;
+    void set_link_layer_type();
```

Comment:
I'm a bit worried about expecting the user to set the link layer type explicitly. For example, I just added support for [Linux Cooked header types](https://github.com/nprint/nprint/blob/eb76192cb0d5dd606da479663d4782084c9f5148/src/io/pcap_parser.cpp#L38) in the file parser itself not the superpacket. Would there ever be a case where traffic in a capture is mixed between wireless and wired?

Reply:
A good question. If we passively monitor the WLAN radio traffic (the way I did), all layers above the 802.11 frame are encrypted and cannot be decrypted easily, unless we know all the passwords for all the Wi-Fis around the monitor devices. 
So, the headers we can actually parse are only "radiotap" + "802.11".

But if we know the password for some wifi connections, it's true that we can decrypt the layers above. Then the layer hierarchy should be "radiotap" + "802.11" + "IP" + ... any headers above IP. 

For the first support, my idea is to make this simple and just consider the first case. Do you have any good design to solve this?

Reply:
Resolved